pygfx/utils/text/_atlas.py

```diff
@@ -231,18 +255,26 @@ def _set_new_glyphs_array(self, size):
             info = self._infos[index]
             x1, y1 = info["origin"]
             w1, h1 = info["size"]
-            if not w1 or not h1:
-                continue  # freed region
+            # An index can have 0 size, for example the
+            # character 32, a non-printable character.
+            # has a glyph that is empty, but it should still
+            # take space and have an index in our atlas.
+            if index in self._free_indices:
+                # freed region
+                continue
```

Comment:
On the main branch, If you add a print statement when width = 0 you can see a few characters are allocated in the waterfall demo.....

Those seem to be various whitespace, one of them 32 (0x20)

Reply:
Nice catch!